### DISCUSSION TITLE
Please be clear about `warning_level`

### DISCUSSION THREAD

**Discussion by mobin-2008** [mesonbuild/meson#11468] [Ideas]
Hi. 
`warning_level` option is documented in https://mesonbuild.com/Builtin-options.html#core-options but this page doesn't give much information about `warning_level`s.
I mean why it's not give me information about every level enable which flags? there is any platform/compiler specific flag? or some older/newer versions of meson enable/disable some additional flags?
I think these things need to be documented.

Regards

**Comment by xclaesse** [comment]
The exact set of flags is compiler specific, but I agree a table mapping level and most common compilers would be nice.

In short:
- 0: no flags
- 1: -Wall
- 2: -Wall -Wextra
- 3: -Wall -Wextra -Wpedantic
- everything: -Weverything (for clang, emulated with list of individual flags with gcc)

**Reply by mobin-2008** [reply]
It's very nice these things be documented. Thanks for your explanations.

**Reply by xclaesse** [reply]
https://github.com/mesonbuild/meson/pull/11470

**Comment by eli-schwartz** [comment]
This was already asked in #7748
